In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import ttest_ind
from scipy.stats import pearsonr

### - Ejercicio 1

Descarga el dataset adjunto de datos oficiales de la UEFA y selecciona un atributo del conjunto de datos. Calcula el p-valor y di si rechaza la hipótesis nula cogiendo un alfa de 5%. 

In [2]:
df = pd.read_csv(r"C:\Users\dpino\Documents\DATA SCIENCE\Jupyter\BOOTCAMP\T5.1\Lligues europees UEFA.csv", sep=";", header=None, engine="python", encoding="latin-1")
pd.set_option('display.max_columns', 500)
df.head(10)

0                1        2     3   4   5  6  7   8   9   10   11     12  \
0  Rk            Squad  Country  LgRk  MP   W  D  L  GF  GA  GD  Pts  Pts/G   
1   1  Manchester City      ENG     1  37  28  6  3  96  24  72   90   2.43   
2   2        Liverpool      ENG     2  36  26  8  2  89  24  65   86   2.39   
3   3      Real Madrid      ESP     1  37  26  7  4  80  31  49   85    2.3   
4   4    Bayern Munich      GER     1  34  24  5  5  97  37  60   77   2.26   
5   5        Paris S-G      FRA     1  37  25  8  4  85  36  49   83   2.24   
6   6            Milan      ITA     1  37  25  8  4  66  31  35   83   2.24   
7   7            Inter      ITA     2  37  24  9  4  81  32  49   81   2.19   
8   8           Napoli      ITA     3  37  23  7  7  71  31  40   76   2.05   
9   9         Dortmund      GER     2  34  22  3  9  85  52  33   69   2.03   

     13    14    15      16         17          18                       19  \
0    xG   xGA   xGD  xGD/90     Last 5  Attendance          Top Team Scorer   
1  86.1  26.8  59.3     1.6  W W W W D       52739     Kevin De Bruyne - 15   
2  84.6  33.1  51.4    1.43  W W W D W       53367       Mohamed Salah - 22   
3    73  45.8  27.2    0.73  W W L W D       40624       Karim Benzema - 27   
4  88.1  37.1    51     1.5  W W L D D       33176  Robert Lewandowski - 35   
5  71.6  38.1  33.4     0.9  W D D D W       41188       Kylian Mbappé - 25   
6    60  34.4  25.6    0.69  W W W W W       42388         Rafael Leão - 11   
7  78.1  38.2  39.9    1.08  W L W W W       41995    Lautaro Martínez - 21   
8  58.5  29.9  28.6    0.77  D L W W W       26681      Victor Osimhen - 14   
9  62.8  43.2  19.6    0.58  W L L W W       41800      Erling Haaland - 22   

                 20  
0        Goalkeeper  
1           Ederson  
2           Alisson  
3  Thibaut Courtois  
4      Manuel Neuer  
5      Keylor Navas  
6      Mike Maignan  
7  Samir Handanovi?  
8      David Ospina  
9      Gregor Kobel

In [3]:
df.columns = df.iloc[0]
df

0   Rk            Squad  Country  LgRk  MP   W   D   L  GF  GA   GD  Pts  \
0   Rk            Squad  Country  LgRk  MP   W   D   L  GF  GA   GD  Pts   
1    1  Manchester City      ENG     1  37  28   6   3  96  24   72   90   
2    2        Liverpool      ENG     2  36  26   8   2  89  24   65   86   
3    3      Real Madrid      ESP     1  37  26   7   4  80  31   49   85   
4    4    Bayern Munich      GER     1  34  24   5   5  97  37   60   77   
..  ..              ...      ...   ...  ..  ..  ..  ..  ..  ..  ...  ...   
94  94         Bordeaux      FRA    20  37   5  13  19  48  89  -41   28   
95  95          Venezia      ITA    20  37   6   8  23  34  69  -35   26   
96  96          Watford      ENG    19  37   6   5  26  33  75  -42   23   
97  97     Norwich City      ENG    20  37   5   7  25  23  79  -56   22   
98  98   Greuther Fürth      GER    18  34   3   9  22  28  82  -54   18   

0   Pts/G    xG   xGA    xGD  xGD/90     Last 5  Attendance  \
0   Pts/G    xG   xGA    xGD  xGD/90     Last 5  Attendance   
1    2.43  86.1  26.8   59.3     1.6  W W W W D       52739   
2    2.39  84.6  33.1   51.4    1.43  W W W D W       53367   
3     2.3    73  45.8   27.2    0.73  W W L W D       40624   
4    2.26  88.1  37.1     51     1.5  W W L D D       33176   
..    ...   ...   ...    ...     ...        ...         ...   
94   0.76  41.3  68.5  -27.2   -0.73  D L L L D       22083   
95    0.7  35.4  70.5  -35.1   -0.95  L L L W D        6705   
96   0.62  37.8  64.1  -26.3   -0.71  L L L D L       20614   
97   0.59    33  73.3  -40.4   -1.09  L L L L D       26826   
98   0.53  30.7  60.3  -29.6   -0.87  D L D L L        7672   

0           Top Team Scorer        Goalkeeper  
0           Top Team Scorer        Goalkeeper  
1      Kevin De Bruyne - 15           Ederson  
2        Mohamed Salah - 22           Alisson  
3        Karim Benzema - 27  Thibaut Courtois  
4   Robert Lewandowski - 35      Manuel Neuer  
..                      ...               ...  
94         Hwang Ui-jo - 11     Benoît Costil  
95         Thomas Henry - 9     Sergio Romero  
96     Emmanuel Dennis - 10        Ben Foster  
97         Teemu Pukki - 11          Tim Krul  
98      Branimir Hrgota - 9   Sascha Burchert  

[99 rows x 21 columns]

Tenemos un Dataset con un conjunto de datos, donde se describe el rendimiento de los equipos de futbol en la liga UEFA. con atributos como, posicion del equipo en la tabla de posiciones(LgRk), Partidos jugados (MP),victorias que un equipo ha logrado (W), Empates (D), Derrotas (L), Goles a favor (GF), Goles en Contra (GA), Diferencia de goles (GD), Puntos (Pts), puntos por partido (Pts/G), Goles esperados (xG), Goles Esperados en Contra (xGA), Diferencia de Goles Esperada (xGD), Diferencia de Goles Esperada por 90 minutos (xGD/90), rendimiento del equipo en sus últimos cinco partidos (Last 5), Asistencia de espectadores (Attendance), el jugador que ha marcado más goles para el equipo (Top Team Scorer), portero titular o principal del equipo (Goalkeeper).

Hipotesis Nula: No existe una diferencia significativa entre el promedio de puntos por partido (Pts/G), entre la liga inglesa y la liga española. 

Hipotesis Alternativa: Hay una diferencia significativa entre la media de puntos por partido (Pts/G), entre la liga ingles y la liga española. 

P. valor: ??

Alfa: 5%. 

In [4]:
df.isnull().sum()

0
Rk                 0
Squad              0
Country            0
LgRk               0
MP                 0
W                  0
D                  0
L                  0
GF                 0
GA                 0
GD                 0
Pts                0
Pts/G              0
xG                 0
xGA                0
xGD                0
xGD/90             0
Last 5             0
Attendance         0
Top Team Scorer    0
Goalkeeper         0
dtype: int64

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99 entries, 0 to 98
Data columns (total 21 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Rk               99 non-null     object
 1   Squad            99 non-null     object
 2   Country          99 non-null     object
 3   LgRk             99 non-null     object
 4   MP               99 non-null     object
 5   W                99 non-null     object
 6   D                99 non-null     object
 7   L                99 non-null     object
 8   GF               99 non-null     object
 9   GA               99 non-null     object
 10  GD               99 non-null     object
 11  Pts              99 non-null     object
 12  Pts/G            99 non-null     object
 13  xG               99 non-null     object
 14  xGA              99 non-null     object
 15  xGD              99 non-null     object
 16  xGD/90           99 non-null     object
 17  Last 5           99 non-null     obje

In [6]:
df['Pts/G'] = pd.to_numeric(df['Pts/G'], errors='coerce')

convertimos tipos de datos de Object a Numerico para poder manipular los datos.

In [7]:
eng = df[df["Country"] == "ENG"]
esp = df[df["Country"] == "ESP"]

In [8]:
print("La liga inglesa tuvo un promedio de puntos por partidos de", eng["Pts/G"].mean())
print("La liga española tuvo un promedio de puntos por partidos de", esp["Pts/G"].mean())

La liga inglesa tuvo un promedio de puntos por partidos de 1.3815000000000002
La liga española tuvo un promedio de puntos por partidos de 1.3524999999999998


In [9]:
t_statistic, p_value = ttest_ind(eng['Pts/G'], esp['Pts/G'], equal_var=False)
print ("t_statistic", t_statistic)
print ("p_value", p_value)
if p_value > 0.05:
    print('No rechazamos la Hipotecis Nula')
else:
    print('Rechazamos la Hipotesis Nula')

t_statistic 0.1989577011043349
p_value 0.8433951242274633
No rechazamos la Hipotecis Nula


Debido a que el valor p es de 0.84 y se encuentra por encima del valor alfa 0.05, no rechazamos la hipotesis nula. lo que significa que, no existe una diferencia significativa entre el promedio de puntos por partido (Pts/G), entre la liga ingles y la liga española. la media de Pts/G para la liga inglesa fue de 1.38, mientras que la media de Pts/G para la liga española es de 1.35. en conclusion promedios similares. 

### - Ejercicio 2

#### Atributo 1
Atributo: Asistencia y Puntos


Hipotesis Nula: No hay una relación significativa entre la cantidad de asistentes (Attendance) y la cantidad de puntos (Pts), en las ligas alemana e italiana.

Hpotesis alternativa: hay una relación significativa entre la cantidad de asistentes (Attendance) y la cantidad de puntos (Pts), en las ligas alemana e italiana.

P. valor: ??

Alpha: 5%

Metodo a usar: Coeficiente de correlación de Pearson


In [10]:
df["Attendance"] = pd.to_numeric(df['Attendance'], errors='coerce') #cambiar object a numerico solo 1 columna 
df[['Attendance', 'Pts']] = df[['Attendance', 'Pts']].apply(pd.to_numeric, errors='coerce') #cambiar object a numerico en mas de una columna  

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99 entries, 0 to 98
Data columns (total 21 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Rk               99 non-null     object 
 1   Squad            99 non-null     object 
 2   Country          99 non-null     object 
 3   LgRk             99 non-null     object 
 4   MP               99 non-null     object 
 5   W                99 non-null     object 
 6   D                99 non-null     object 
 7   L                99 non-null     object 
 8   GF               99 non-null     object 
 9   GA               99 non-null     object 
 10  GD               99 non-null     object 
 11  Pts              98 non-null     float64
 12  Pts/G            98 non-null     float64
 13  xG               99 non-null     object 
 14  xGA              99 non-null     object 
 15  xGD              99 non-null     object 
 16  xGD/90           99 non-null     object 
 17  Last 5           9

In [12]:
ger = df[df["Country"] == "GER"]
ita = df[df["Country"] == "ITA"]

In [13]:
print("La liga alemana tuvo una asistencia de", ger['Attendance'].sum())
print("La liga italina tuvo una asistencia de", ita['Attendance'].sum())

La liga alemana tuvo una asistencia de 378451.0
La liga italina tuvo una asistencia de 341249.0


In [14]:
print("La liga alemana tuvo un total de puntos de", ger['Pts'].sum())
print("La liga italina tuvo un total de puntos de", ita['Pts'].sum())

La liga alemana tuvo un total de puntos de 845.0
La liga italina tuvo un total de puntos de 1009.0


In [15]:
corr_ale, p_value_ale = pearsonr(ger['Attendance'], ger['Pts'])
print(f"Correlación para la liga alemana: {corr_ale}")
print(f"P-valor para la liga alemana: {p_value_ale}")
if p_value_ale > 0.05:
    print('No rechazamos la Hipotecis Nula')
else:
    print('Rechazamos la Hipotesis Nula')

Correlación para la liga alemana: 0.5360159821541407
P-valor para la liga alemana: 0.02185188581321732
Rechazamos la Hipotesis Nula


Para la liga Alemana rechazamos la Hipotesis Nula,  No hay una relación significativa entre la cantidad de asistentes (Attendance) y la cantidad de puntos (Pts); debido a que si existe una relacion entre la cantidad de asistencia a los partidos y la cantidad de puntos obtenidos por el equipo. 

In [16]:
corr_ita, p_ita = pearsonr(ita['Attendance'], ita['Pts'])
print(f"\nCorrelación para la liga italiana: {corr_ita}")
print(f"P-valor para la liga italiana: {p_ita}")
if p_value > 0.05:
    print('No rechazamos la Hipotecis Nula')
else:
    print('Rechazamos la Hipotesis Nula')


Correlación para la liga italiana: 0.8018123086268638
P-valor para la liga italiana: 2.1276016791645627e-05
No rechazamos la Hipotecis Nula


Para la liga italiana No rechazamos la Hipotesis Nula, No hay una relación significativa entre la cantidad de asistentes (Attendance) y la cantidad de puntos (Pts). por tanto podriamos decir que la cantidad de asitentes a los partidos no determina,la cantidad de puntos por partidos. 

En conclusion general: la cantidad de asistentes a la liga alemana fue de 378451 y un total de puntos de 845, mientras que la liga italiana tuvo una asistencia de 341249 menos a la alemana y 1009 puntos, mayor a la alemana. podemos ver que para algunas ligas si es importante la asistencia a los partidos, pero para otros es impotante el apoyo de sus fanaticos.

### - Ejercicio 3

Atributo 1
Atributo: Partidos jugados (MP) - victorias que un equipo ha logrado (W) - los Puntos (Pts)

Hipotesis Nula: No hay una relación significativa entre el promedio de, los Partidos jugados (MP), las victorias que un equipo ha logrado (W) y los Puntos (Pts), en las ligas España, Alemania e Inglaterra. 

Hipotesis alternativa: hay una relación significativa entre el promedio de, los Partidos jugados (MP), las victorias que un equipo ha logrado (W) y los Puntos (Pts), en las ligas España, Alemania e Inglaterra.

P. valor: ??

Alpha: 5%

Metodo a usar: Prueba de análisis de varianza (ANOVA)

In [23]:
esp = df[df["Country"] == "ESP"]
ger = df[df["Country"] == "GER"]
eng = df[df["Country"] == "ENG"]

In [30]:
print("La liga española tuvo un promedio de partidos jugados de", esp["MP"].mean())
print("La liga alemana tuvo un promedio de partidos jugados de", ger["MP"].mean())
print("La liga inglesa tuvo un promedio de partidos jugados de", eng["MP"].mean())

La liga española tuvo un promedio de partidos jugados de 1.8686868686868686e+38
La liga alemana tuvo un promedio de partidos jugados de 1.9079685746352414e+34
La liga inglesa tuvo un promedio de partidos jugados de 1.868181868186869e+38


In [31]:
print("La liga española tuvo un promedio de victorias que el equipo ha logrado de", esp["W"].mean())
print("La liga alemana tuvo un promedio de victorias que el equipo ha logrado de", ger["W"].mean())
print("La liga inglesa tuvo un promedio de victorias que el equipo ha logrado de", eng["W"].mean())

La liga española tuvo un promedio de victorias que el equipo ha logrado de 1.3106008595857572e+32
La liga alemana tuvo un promedio de victorias que el equipo ha logrado de 1.345662064528563e+30
La liga inglesa tuvo un promedio de victorias que el equipo ha logrado de 1.4131010605808075e+33


In [34]:
print("La liga española tuvo un promedio de puntos de", esp["Pts"].mean())
print("La liga alemana tuvo un promedio de puntos de", ger["Pts"].mean())
print("La liga inglesa tuvo un promedio de puntos de", eng["Pts"].mean())

La liga española tuvo un promedio de puntos de 50.05
La liga alemana tuvo un promedio de puntos de 46.94444444444444
La liga inglesa tuvo un promedio de puntos de 50.45


Aplicacion del metodo para cada liga

In [17]:
from scipy.stats import f_oneway
data1 = esp['MP']
data2 = esp['W']
data3 = esp['Pts']
stat, p = f_oneway(data1, data2, data3)
print('stat=%.3f, p=%.3f' % (stat, p))
if p > 0.05:
 print('No se rechaza la hipótesis nula')
else:
 print('Se rechaza la hipótesis nula')

stat=80.260, p=0.000
Se rechaza la hipótesis nula


por tanto, la hipotesis  nula queda rechazada, puesto que hay una relación significativa entre el promedio de, los Partidos jugados (MP) (1.868), las victorias que un equipo ha logrado (W)(1.310 ) y los Puntos (Pts)(50.05), en las ligas España. 

In [18]:
from scipy.stats import f_oneway
data1 = ger['MP']
data2 = ger['W']
data3 = ger['Pts']
stat, p = f_oneway(data1, data2, data3)
print('stat=%.3f, p=%.3f' % (stat, p))
if p > 0.05:
 print('No se rechaza la hipótesis nula')
else:
 print('Se rechaza la hipótesis nula')

stat=63.700, p=0.000
Se rechaza la hipótesis nula


por tanto, la hipotesis  nula queda rechazada, puesto que hay una relación significativa entre el promedio de, los Partidos jugados (MP) (1.90), las victorias que un equipo ha logrado (W)(1.34) y los Puntos (Pts)(46.9), en las ligas Alemana. 

In [35]:
from scipy.stats import f_oneway
data1 = eng['MP']
data2 = eng['W']
data3 = eng['Pts']
stat, p = f_oneway(data1, data2, data3)
print('stat=%.3f, p=%.3f' % (stat, p))
if p > 0.05:
 print('No se rechaza la hipótesis nula')
else:
 print('Se rechaza la hipótesis nula')

stat=53.764, p=0.000
Se rechaza la hipótesis nula


por tanto, la hipotesis  nula queda rechazada, puesto que hay una relación significativa entre el promedio de, los Partidos jugados (MP) (1.868), las victorias que un equipo ha logrado (W)(1.41) y los Puntos (Pts)(50.45), en las ligas Inglesa. 

conclusion general: las medias de los tres atributos no tienen relacion significativa entre si, en niguna de las tres muestras de ligas.  